src/app/debug/table/table.component.ts

```diff
@@ -662,4 +667,32 @@ export class TableComponent implements OnInit, OnDestroy {
     }
     return result;
   }
+
+  getMetadata(report: Report, field: string): string {
+    return report[field as keyof Report];
+  }
+
+  sortData(sort: Sort): any {
+    if (!sort.active || sort.direction === '') {
+      this.sortedReportMetadata = [...this.tableSettings.reportMetadata];
+      return;
+    }
+    this.sortedReportMetadata.sort((a: Report, b: Report): number => {
+      const isAsc: boolean = sort.direction === 'asc';
+      const headersA: [string, string][] = Object.entries(a);
+      const headersB: [string, string][] = Object.entries(b);
+      for (const [index, element] of headersA.entries()) {
+        if (Number(sort.active) === index) {
+          return compare(element[1], headersB[index][1], isAsc);
+        }
+      }
+      return 0;
+    });
+  }
+
+  protected readonly String = String;
+}
+
+function compare(a: number | string, b: number | string, isAsc: boolean): number {
+  return (a < b ? -1 : 1) * (isAsc ? 1 : -1);
```

Comment:
Is it me or does this seem overly complex? Have you looked at [ngBootstrap tables](https://ng-bootstrap.github.io/#/components/table/examples#sortable)?
Also, why is this function outside of the component class?

Reply:
> Also, why is this function outside of the component class?

It was a copy and paste from helper service and angular docs example showed the same thing. I changed it so it is now in the component.

> Is it me or does this seem overly complex? Have you looked at [ngBootstrap tables](https://ng-bootstrap.github.io/#/components/table/examples#sortable)?

I took a look at it and I find that more unreadable that the current code, but I might be biased since I just worked on it. The example for sorting seems like you would write a lot of code that the matSort already does for you. Would be great to hear your opinion on this.

Reply:
It might be easier to just use [angular material table](https://material.angular.io/components/table/overview#sorting)

Reply:
@MatthijsSmets Replaced the old table with the angular material table. This resulted in not needing a sorting method anymore.